lib/src/char_encodings.dart

```diff
@@ -24,7 +20,7 @@ bool hasUtf8Bom(List<int> bytes, [int offset = 0, int length]) {
 Iterable<int> decodeBytes(String encoding, List<int> bytes,
     [int offset = 0,
     int length,
-    int replacementCodepoint = UNICODE_REPLACEMENT_CHARACTER_CODEPOINT]) {
+    int replacementCodepoint = unicodeReplacementCharacterCodepoint]) {
```

Comment:
should we instead replace how the decoder function is used so that users can provide a different implementation?

In other words, instead of removing the support for some encodings, we can take them off the critical path, but continue to let users customize how they use this package to get the same behavior.

If we believe the other encodings are never used, that's a different story.

Reply:
There's no evidence that anyone uses this package with anything other than ASCII/UTF-8.

I ran a presubmit internally – zero breaks.

We can always add such an abstraction back if needed.

Reply:
Yeah Siggi, I was concerned too, and spend some time trying to understand what Kevin meant by "deprecated encodings". I found:

https://html.spec.whatwg.org/multipage/infrastructure.html#encoding-terminology
https://encoding.spec.whatwg.org/#encodings

So it does seem that all other encodings are deprecated (marked "Legacy"). I'm not sure how often deprecated encodings occur in real life HTML servers/documents, but it seems reasonable to remove support for them, given that.